sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -164,35 +164,52 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase {
         s"side of the join. The $side-side columns: [${plan.output.map(_.name).mkString(", ")}]")
   }
 
+  private def unresolvedAttributeSuggestion(errorClass: String, candidates: Seq[String]): String = {
+    if (candidates.isEmpty) {
+      ""
+    } else {
+      val candidateIds = candidates.map(toSQLId).mkString(", ")
+      if (errorClass == "UNRESOLVED_MAP_KEY") {
+        s" Otherwise did you mean one of the following column(s)? [$candidateIds]"
```

Comment:
Could you imagine that we will translate error messages in error-classes.json to local languages. It would be better if we put the text of error message to the json file, and parametrise it by some values that don't require translation.

Is it possible to create sub-classes w/ suggestions in the json file?

Reply:
That is such a great comment! Thank you, I added sub classes for respective errors. We also have some other instances of hard coded strings, I created a ticket ([SPARK-40236](https://issues.apache.org/jira/browse/SPARK-40236)) and fix those.